commit 8fa947e8c3f2faa7a20ff0572e0b7c3e4cd3f2ea (refs/changes/57/1657/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-03-14T12:18:02+01:00 (2 years, 7 months ago)
    
    TOOLS-1719 sdcadm create should fail gracefully if a service hasn't got image uuid

diff --git a/lib/cli/do_create.js b/lib/cli/do_create.js
index 00e6a54..6f3e3de 100644
--- a/lib/cli/do_create.js
+++ b/lib/cli/do_create.js
@@ -111,6 +111,16 @@ Create.prototype.execute = function cExecute(opts, args, cb) {
                 change.image = svcFromName[service].params.image_uuid;
             }
 
+            /*
+             * TOOLS-1719: Fail graceful if service doesn't have an associated
+             * image, instead of downloading the latest available image for
+             * service
+             */
+            if (!change.image) {
+                return next(new errors.ValidationError(util.format(
+                    'Missing image_uuid for service %s in SAPI.', service)));
+            }
+
             if (!opts.server) {
                 return next(new errors.UsageError(
                     'Must specify server uuid or hostname'));
